build/sync/plan.yml

```diff
@@ -0,0 +1,204 @@
+checks:
+  - type: repo_is_clean
+    params:
+      repo: template
+  - type: repo_is_clean
+    params:
+      repo: plugin
+paths:
+  - path: go.mod
+    actions:
+      - type: overwrite_file
+        params:
+          create: false
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: build
+    actions:
+      - type: overwrite_directory
+        params:
+          create: true
+  - path: Makefile
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: .editorconfig
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: webapp/.babelrc
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: webapp/webpack.config.js
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: webapp/tsconfig.json
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: webapp/.eslintrc.json
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: webapp/src/index.js
+    actions:
+      - type: overwrite_file
+        params:
+          create: true
+        conditions:
+          - type: exists
+            params:
+              repo: plugin
+          - type: file_unaltered
+            params:
+              repo: plugin
+              reference-repo: template
+  - path: server/configuration.go
```

Comment:
That seems more like a checker/linter type tool - I think it makes sense to make that a separate command, instead of integrating it with the sync tool.